src/main/java/domain/LottoNumber.java

```diff
@@ -0,0 +1,44 @@
+package domain;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+public class LottoNumber {
+    public static final int MIN_VALUE = 1;
+    public static final int MAX_VALUE = 45;
+    private static final String LOTTO_NUMBER_RANGE_ERROR_MESSAGE = String.format("로또 번호는 %d~%d 사이로 입력해주세요.",
+            MIN_VALUE, MAX_VALUE);
+
+    private static final Map<Integer, LottoNumber> lottoNumberCache = new HashMap<>() {{
+        IntStream.rangeClosed(MIN_VALUE, MAX_VALUE)
+                .boxed()
+                .map(LottoNumber::new)
+                .forEach(lottoNumber -> put(lottoNumber.getValue(), lottoNumber));
+    }};
```

Comment:
```suggestion
    private static final Map<Integer, LottoNumber> lottoNumberCache = new HashMap<>() {{
        IntStream.rangeClosed(MIN_VALUE, MAX_VALUE)
                .mapToObj(LottoNumber::new)
                .forEach(lottoNumber -> put(lottoNumber.getValue(), lottoNumber));
    }};
```

(참고) `Collectors.toMap()`을 이용할 수도 있어요!